ledger/src/shred/merkle.rs

```diff
@@ -953,6 +956,9 @@ pub(super) fn recover(
         .map(Shred::merkle_node)
         .collect::<Result<_, _>>()?;
     let tree = make_merkle_tree(nodes);
+    if tree.last() != Some(&merkle_root) {
+        return Err(Error::InvalidMerkleRoot);
```

Comment:
For my understanding:
Since we verified that all the shreds used in recovery have the same signature is there a setup that would allow a leader to give us a merkle tree with a different root here?

Reply:
I think we only do `debug_assert` on signatures being equal.
But below at line 968, we do verify that the Merkle proofs are the same as the reconstructed tree.
Might actually make sense to also always check if signatures are equal since that is cheap and can be done earlier.

I think it is potentially possible that for the same data shreds the leader generates two sets of coding shreds: sets A and B.
Then on the receiving end, the signature (and also the Merkle tree) you get depends on if you recover with coding shreds from set A or set B.
But either case the signature verifies if we get the same Merkle root as the coding shred that we used because that one is sig-verified.